### PR TITLE
backport Gluon modules Updater

### PR DESCRIPTION
**Pull request by T0biii**
backport: https://github.com/freifunk-gluon/gluon/blob/66d3b1749b0d2d57f579f4e28ffe80bb2b8d5272/scripts/update-modules.sh

--- stable ---
Run CI stable: https://github.com/T0biiias/site-ffm5/actions/runs/10013543786
PR stable: https://github.com/T0biiias/site-ffm5/pull/18

--- legacy ---
Run CI legacy : https://github.com/T0biiias/site-ffm5/actions/runs/10015876445
PR legacy : https://github.com/T0biiias/site-ffm5/pull/20

only this commit is missing: https://github.com/freifunkMUC/site-ffm/pull/448/commits/85238b57d5b29c91655f55345b065f529f2c4b60

<details>
<summary>old</summary>
<br>

--- stable ---
Test (downgrade module commit) stable: https://github.com/T0biiias/site-ffm5/commit/376160cb32816c6edaddbaf7864b7de6a3510656
Run CI stable: https://github.com/T0biiias/site-ffm5/actions/runs/9735952347
PR stable: https://github.com/T0biiias/site-ffm5/pull/8  

--- legacy ---
Test (downgrade module commit) legacy: https://github.com/T0biiias/site-ffm5/commit/1c4265ddad2968923faa44f660843c347087e6cb
Run CI legacy: https://github.com/T0biiias/site-ffm5/actions/runs/9735963650
PR legacy: https://github.com/T0biiias/site-ffm5/pull/9 

</details>

### CODE REVIEW

File: contrib/actions/update-modules.sh [hunk 1]
```diff
@@ -1,39 +1,68 @@
...
-# file that contains the module information
-MODULES_FILE="modules"
+set -eo pipefail
 
-if ((BASH_VERSINFO[0] < 4)); then
-    echo "This script requires Bash 4.0 or above."
-    exit 1
-fi
+# move to basedir, in case the script is not executed via `make update-modules`
+cd "../../" || exit 1
```
Comment (T0biii) [inline contrib/actions/update-modules.sh]: @grische the original from the Gluon Script doesn't worked for me in the CI. Maybe you got a better command than this for me? (with this it only works atm in console when you are in the same folder as the script)

File: contrib/actions/update-modules.sh [hunk 2]
```diff
@@ -1,39 +1,68 @@
...
 }
 
-for repo in $(get_all_repo_names "${MODULES_FILE}"); do
-    REPO_URL=$(grep "^PACKAGES_${repo^^}_REPO" "${MODULES_FILE}" | get_value_from_line)
-    REPO_COMMIT=$(grep "^PACKAGES_${repo^^}_COMMIT" "${MODULES_FILE}" | get_value_from_line)
-    REPO_BRANCH=$(grep "^PACKAGES_${repo^^}_BRANCH" "${MODULES_FILE}" | get_value_from_line)
+LOCAL_BRANCH=$(git branch --show-current)
+[[ $LOCAL_BRANCH != update-modules-* ]] && LOCAL_BRANCH="update-modules-$LOCAL_BRANCH-$(date +%s)"
+
+for MODULE in "PACKAGES_COMMUNITY"; do
```
Comment (T0biii) [inline contrib/actions/update-modules.sh]: atm only on legacy we have more packages Repo than community (CI is failing because of this, Loop only once)

**Review by T0biii — COMMENTED** [review]
@grische maybe you got some ideas how to handle the open issues

**Review by T0biii — COMMENTED** [review]
@grische maybe you got some ideas how to handle the open issues

File: .github/workflows/bump-modules.yml [hunk 3]
```diff
@@ -32,7 +32,6 @@ jobs:
             - Auto-generated by [create-pull-request][1]
```
Comment (grische) [inline .github/workflows/bump-modules.yml]: Do you like `branche` over `branch`? 😉

File: contrib/actions/update-modules.sh (hunk 2 above)
Comment (grische) [reply contrib/actions/update-modules.sh]: Can't you just use the beautiful loop we had before?
```sh
...
for repo in $(get_all_repo_names "${MODULES_FILE}"); do
...
```

Comment (grische) [reply contrib/actions/update-modules.sh]: Found a much better solution 🙂 
https://github.com/freifunkMUC/site-ffm/pull/448#discussion_r1661442864

File: contrib/actions/update-modules.sh [hunk 4]
```diff
@@ -1,39 +1,63 @@
...
-    REPO_URL=$(grep "^PACKAGES_${repo^^}_REPO" "${MODULES_FILE}" | get_value_from_line)
-    REPO_COMMIT=$(grep "^PACKAGES_${repo^^}_COMMIT" "${MODULES_FILE}" | get_value_from_line)
-    REPO_BRANCH=$(grep "^PACKAGES_${repo^^}_BRANCH" "${MODULES_FILE}" | get_value_from_line)
 
-    # Get newest commit of the repo
-    NEW_COMMIT=$(git ls-remote --heads "${REPO_URL}" "${REPO_BRANCH}" | grep -oE '[0-9a-f]{40}')
+for MODULE in "PACKAGES_COMMUNITY"; do
+	_REMOTE_URL=${MODULE}_REPO
+	_REMOTE_BRANCH=${MODULE}_BRANCH
+	_LOCAL_HEAD=${MODULE}_COMMIT
```
Comment (grische) [inline contrib/actions/update-modules.sh]:
```suggestion
for MODULE in ${GLUON_SITE_FEEDS}; do
	_REMOTE_URL=PACKAGES_${MODULE^^}_REPO
	_REMOTE_BRANCH=PACKAGES_${MODULE^^}_BRANCH
	_LOCAL_HEAD=PACKAGES_${MODULE^^}_COMMIT
```

File: .github/workflows/bump-modules.yml [hunk 5]
```diff
@@ -28,11 +28,10 @@ jobs:
         with:
           title: '[Update] ${{ github.event.inputs.branch }} modules'
           body: |
-            Update modules for ${{ github.event.inputs.branch }} branche
-            - Auto-generated by [create-pull-request][1]
+            Update modules for ${{ github.event.inputs.branch }} branch
```
Comment (grische) [inline .github/workflows/bump-modules.yml]:
```suggestion
            Update modules on branch ${{ github.event.inputs.branch }}
```

File: contrib/actions/update-modules.sh [hunk 6]
```diff
@@ -1,39 +1,63 @@
...
+
+	# prepare the commit message
+	# shellcheck disable=SC2001
+	MODULE=$(echo "${MODULE,,}" | sed 's/packages_//')
+	TITLE="modules: update ${MODULE}"
+	MESSAGE="$(mktemp)"
+	{
+		echo "${TITLE}"
+		printf '\n\n'
+		git -C "${CHECKOUT}" log --oneline --no-decorate --no-merges "${LOCAL_HEAD}..${REMOTE_HEAD}" | cat
```
Comment (T0biii) [inline contrib/actions/update-modules.sh:52]: test
```suggestion

	# clone the target branch
	git clone --bare "${REMOTE_URL}" --branch="${REMOTE_BRANCH}" "${CHECKOUT}"
        # i would delete all
	# prepare the commit message
	# shellcheck disable=SC2001
	MODULE=$(echo "${MODULE,,}" | sed 's/packages_//')
	TITLE="modules: update ${MODULE}"
	MESSAGE="$(mktemp)"
	{
		echo "${TITLE}"
		printf '\n\n'
		git -C "${CHECKOUT}" log --oneline --no-decorate --no-merges "${LOCAL_HEAD}..${REMOTE_HEAD}" | cat
```

**Comment by grische** [comment]
@T0biii LGTM in general. Any further blockers? Do you maybe have some updated example runs?

**Comment by T0biii** [comment]
@grische the only "blocker" is some updated example run :)

File: .github/workflows/bump-modules.yml [hunk 7]
```diff
@@ -11,13 +11,19 @@ permissions:
...
       - name: Clone Firmware
         uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
 
+      - name: Configure Git User and E-Mail
+        run: git config --global user.name "${{ env.COMMIT_NAME }}" && git config --global user.email "${{ env.COMMIT_EMAIL }}"
+        
       - name: Get update branch name
         id: branch-name
-        run: echo "branch-name=update-modules-${{ github.event.inputs.branch }}-$(date +%s)" >> $GITHUB_OUTPUT
+        run: echo "branch-name=update-modules-${{ github.ref_name }}-$(date +%s)" >> $GITHUB_OUTPUT
```
Comment (grische) [inline .github/workflows/bump-modules.yml:26]: Why is this needed? Does this still work with workflow_dispatch?

**Comment by T0biii** [comment]
@grische now it is ready for review

File: .github/workflows/bump-modules.yml [hunk 8]
```diff
@@ -11,13 +11,19 @@ permissions:
 jobs:
   update-Modules:
     runs-on: ubuntu-22.04
+    env:
+      COMMIT_NAME: github-actions[bot] 
+      COMMIT_EMAIL: 41898282+github-actions[bot]@users.noreply.github.com
```
Comment (grische) [inline .github/workflows/bump-modules.yml:16]: The `[]` seem to make this an invalid email. 




We can instead just use a generic ffmuc mail and then give the bot our own name.

**Review by grische — COMMENTED** [review]
Are there any changes in comparison to the upstream version on this file?

File: .github/workflows/bump-modules.yml (hunk 7 above)
Comment (T0biii) [reply .github/workflows/bump-modules.yml:26]: this is needed otherwise there is no branch name see https://github.com/T0biiias/site-ffm5/pull/18 there is missing this commit. 
workflow_dispatch works with this just fine

File: .github/workflows/bump-modules.yml (hunk 8 above)
Comment (T0biii) [reply .github/workflows/bump-modules.yml:16]: i would stick to the github actions e-mail for the profil Picture:
image
also it seems to work just fine with the `[]` also copied from the create-pull-request workflow:
https://github.com/peter-evans/create-pull-request/blob/d884c1e53f6f67d5ba26f3abc0159ad6c12c16d8/action.yml#L27

Comment (grische) [reply .github/workflows/bump-modules.yml:16]: OK, apparently they really use the `[...]`. 
I just checked and our dependabot has a different ID:
```
commit 764ef92afb5531ab81e2349d9a4d2644131eea04
Author:     dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
AuthorDate: Wed Jun 19 07:33:52 2024 +0000
Commit:     GitHub <noreply@github.com>
CommitDate: Wed Jun 19 07:33:52 2024 +0000
```

Comment (grische) [reply .github/workflows/bump-modules.yml:16]: The current ID redirects to the generic actions page :+1:

**Review by grische — APPROVED** [review]
